riak-driver/riak-key-value/riak-key-value-api/src/main/java/org/jnosql/diana/riak/key/RiakKeyValueEntityManager.java

```diff
@@ -0,0 +1,189 @@
+
+import com.basho.riak.client.api.RiakClient;
+import com.basho.riak.client.api.cap.UnresolvedConflictException;
+import com.basho.riak.client.api.commands.kv.DeleteValue;
+import com.basho.riak.client.api.commands.kv.FetchValue;
+import com.basho.riak.client.api.commands.kv.FetchValue.Response;
+import com.basho.riak.client.api.commands.kv.StoreValue;
+import com.basho.riak.client.core.query.Namespace;
+import com.google.gson.Gson;
+
+public class RiakKeyValueEntityManager implements BucketManager {
+
+
+	private final RiakClient client;
+	
+	private final Gson gson;
+	
+	private final Namespace nameSpace;
+	
+	public RiakKeyValueEntityManager(RiakClient client, Gson gson, Namespace nameSpace) {
+		this.client = client;
+		this.gson = gson;
+		this.nameSpace = nameSpace;
+	}
+
+	@Override
+	public <K, V> void put(K key, V value) throws NullPointerException {
+		put(KeyValueEntity.of(key,value));
+	}
+
+	@Override
+	public <K> void put(KeyValueEntity<K> entity) throws NullPointerException {
+		put(entity,Duration.ZERO);
+	}
+
+	@Override
+	public <K> void put(KeyValueEntity<K> entity, Duration ttl)
+			throws NullPointerException, UnsupportedOperationException {
+
+		K key = entity.getKey();
+		Value value = entity.getValue();
+
+		StoreValue storeValue = createStoreValue(key,value,nameSpace,ttl);
+		
+		try {
+			client.execute(storeValue);
+		} catch (ExecutionException e) {
+			throw new DianaRiakException(e.getMessage(),e);
+		} catch (InterruptedException e) {
+			throw new DianaRiakException(e.getMessage(),e);
+		}
+	}
+
+	@Override
+	public <K> void put(Iterable<KeyValueEntity<K>> entities) throws NullPointerException {
+		StreamSupport.stream(entities.spliterator(),false).forEach(this::put);
+	}
+
+	@Override
+	public <K> void put(Iterable<KeyValueEntity<K>> entities, Duration ttl)
+			throws NullPointerException, UnsupportedOperationException {
+		
+		StreamSupport.stream(entities.spliterator(),false).forEach(e -> put(e,ttl));
+	}
+
+	@Override
+	public <K> Optional<Value> get(K key) throws NullPointerException {
+		
+		if(StringUtils.isBlank(key.toString())){
+			throw new DianaRiakException("The Key is irregular",new IllegalStateException());
+		}
+
+		FetchValue fetchValue = createFetchValue(nameSpace, key);
+		
+		try {
+			FetchValue.Response response = client.execute(fetchValue);
+			String valueFetch = response.getValue(String.class);
+			if(StringUtils.isNoneBlank(valueFetch))
+				 return Optional.of(RiakValue.of(gson,valueFetch));
+	    
+		} catch (UnresolvedConflictException e) {
+			throw new DianaRiakException(e.getMessage(),e);
+		} catch (ExecutionException e) {
+			throw new DianaRiakException(e.getMessage(),e);
+		} catch (InterruptedException e) {
+			throw new DianaRiakException(e.getMessage(),e);
+		}
+	     
+		return Optional.empty();
+	}
+
+	@Override
+	public <K> Iterable<Value> get(Iterable<K> keys) throws NullPointerException {
+		
+	/* Stream Version , is so ugly , but i waiting feedback for implemententions
```

Comment:
What version of implemetation you prefer?

https://gist.github.com/brenopessoa/7aaceaa56dfeefd41a8f1754e0ca6856

or

https://gist.github.com/brenopessoa/e42d0642d615e2607ba1813d6c43bb23